basedecorator.rb

```diff
@@ -0,0 +1,12 @@
+require './nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
```

Comment:
- [ ] it would be better to add `attr_accessor :nameable` so that you can set and get nameable